Hey @aequitas, I tested if it works again using the github repository in a development environment this time instead of overriding via custom_components and the code worked beautifully!
I tried adding a test but something is still wrong. On line 407 I always get:
```
>       assert hass.states.get('light.toggle_test').state == 'on'
E       AssertionError: assert 'off' == 'on'
E         - off
E         + on
```
I don't know why, since the state change worked without any issues in the frontend and I could not find any other errors... I'd be glad if you could have a look and tell me if you can find anything.

--- a/homeassistant/components/light/rflink.py
+++ b/homeassistant/components/light/rflink.py
@@ -267,10 +267,9 @@ class ToggleRflinkLight(SwitchableRflinkDevice, Light):
 
         command = event['command']
         if command == 'on':
-            if self._state is (STATE_UNKNOWN or False):
-                self._state = True
-            else:
-                self._state = False
+            # if the state is unknown or false, it gets set as true
+            # if the state is true, it gets set as false
+            self._state = self._state is (STATE_UNKNOWN or False)
 
     @asyncio.coroutine
     def async_turn_on(self, **kwargs):

--- a/homeassistant/components/light/rflink.py
+++ b/homeassistant/components/light/rflink.py
@@ -15,8 +15,8 @@ from homeassistant.components.rflink import (
     CONF_IGNORE_DEVICES, CONF_NEW_DEVICES_GROUP, CONF_SIGNAL_REPETITIONS,
     DATA_DEVICE_REGISTER, DATA_ENTITY_LOOKUP, DEVICE_DEFAULTS_SCHEMA, DOMAIN,
     EVENT_KEY_COMMAND, EVENT_KEY_ID, SwitchableRflinkDevice, cv, vol)
-from homeassistant.const import CONF_NAME, CONF_PLATFORM, CONF_TYPE, STATE_UNKNOWN
-
+from homeassistant.const import (
+    CONF_NAME, CONF_PLATFORM, CONF_TYPE, STATE_UNKNOWN)
 DEPENDENCIES = ['rflink']
 
 _LOGGER = logging.getLogger(__name__)

--- a/homeassistant/components/light/rflink.py
+++ b/homeassistant/components/light/rflink.py
@@ -15,7 +15,7 @@ from homeassistant.components.rflink import (
     CONF_IGNORE_DEVICES, CONF_NEW_DEVICES_GROUP, CONF_SIGNAL_REPETITIONS,
     DATA_DEVICE_REGISTER, DATA_ENTITY_LOOKUP, DEVICE_DEFAULTS_SCHEMA, DOMAIN,
     EVENT_KEY_COMMAND, EVENT_KEY_ID, SwitchableRflinkDevice, cv, vol)
-from homeassistant.const import CONF_NAME, CONF_PLATFORM, CONF_TYPE
+from homeassistant.const import CONF_NAME, CONF_PLATFORM, CONF_TYPE, STATE_UNKNOWN
 
 DEPENDENCIES = ['rflink']
 
@@ -267,7 +267,10 @@ class ToggleRflinkLight(SwitchableRflinkDevice, Light):
 
         command = event['command']
         if command == 'on':
-            self._state = not self._state
+            if self._state is (STATE_UNKNOWN or False):
+                self._state = True
+            else:
+                self._state = False
 
     @asyncio.coroutine
     def async_turn_on(self, **kwargs):

--- a/homeassistant/components/rflink.py
+++ b/homeassistant/components/rflink.py
@@ -319,10 +319,9 @@ class RflinkCommand(RflinkDevice):
 
         elif command == 'toggle':
             cmd = 'on'
-            if self._state is (STATE_UNKNOWN or False):
-                self._state = True
-            else:
-                self._state = False
+            # if the state is unknown or false, it gets set as true
+            # if the state is true, it gets set as false
+            self._state = self._state is (STATE_UNKNOWN or False)
 
         # Send initial command and queue repetitions.
         # This allows the entity state to be updated quickly and not having to

--- a/homeassistant/components/rflink.py
+++ b/homeassistant/components/rflink.py
@@ -319,7 +319,10 @@ class RflinkCommand(RflinkDevice):
 
         elif command == 'toggle':
             cmd = 'on'
-            self._state = not self._state
+            if self._state is (STATE_UNKNOWN or False):
+                self._state = True
+            else:
+                self._state = False
 
         # Send initial command and queue repetitions.
         # This allows the entity state to be updated quickly and not having to

